gateway/handlers/forwarding_proxy.go

```diff
@@ -140,6 +140,30 @@ func copyHeaders(destination http.Header, source *http.Header) {
 	}
 }
 
+type PrometheusServiceNotifier struct {
+	ServiceMetrics *metrics.ServiceMetricOptions
+}
+
+// Notify about service metrics
+func (psn PrometheusServiceNotifier) Notify(method string, URL string, originalURL string, statusCode int, duration time.Duration) {
+	code := fmt.Sprintf("%d", statusCode)
+	psn.ServiceMetrics.Counter.WithLabelValues(code).Inc()
+
+	psn.ServiceMetrics.Histogram.WithLabelValues(method, urlToLabel(URL), code).Observe(duration.Seconds())
+}
+
+var invalidChars = regexp.MustCompile(`[^a-zA-Z0-9]+`)
```

Comment:
I have seen labels with `/`, I have also never had an issue querying and filtering on those labels. Is there a reference for which characters cause issues?

Reply:
Pretty sure `/` wasn't allowed in labels.

Reply:
![image](https://user-images.githubusercontent.com/891889/50723084-2965dd00-10d9-11e9-9fa3-54d3454fd50f.png)

Reply:
I have a prometheus scraping metrics with slashes right now (the space is accidental and being fixed).  But it seems like labels are really flexible

Reply:
@stefanprodan WDYT about slashes?

Reply:
I'm using this to converts gorilla mux routes into Prom labels: https://github.com/stefanprodan/k8s-podinfo/blob/master/pkg/api/metrics.go#L65